rrfs-test/validated_yamls/templates/obtype_config/adpsfc_airTemperature_187.yaml

```diff
@@ -0,0 +1,194 @@
+     - obs space:
+         name: adpsfc_airTemperature_187
+         distribution:
+           name: "@DISTRIBUTION@"
+           halo size: 100e3
+         obsdatain:
+           engine:
+             type: H5File
+             obsfile: "@OBSFILE@"
+         obsdataout:
+           engine:
+             type: H5File
+             obsfile: hofx_adpsfc_airTemperature_187.nc4
+             allow overwrite: true
+         io pool:
+           max pool size: 1
+         observed variables: [airTemperature]
+         simulated variables: [airTemperature]
+
+       obs operator:
+         name: Composite
+         components:
+         - name: VertInterp
+           variables:
+           - name: airTemperature
+           vertical coordinate: air_pressure
+           observation vertical coordinate: pressure
+           observation vertical coordinate group: MetaData
+           interpolation method: log-linear
+
+       obs error:
+         covariance model: diagonal
+
+       obs localizations:
+         - localization method: Horizontal Gaspari-Cohn
+           lengthscale: 300e3 # orig
+
+       obs filters:
+         # ------------------
+         # airTemperature (187)
+         # ------------------
+         # Reject all obs with ObsType /= 187, QualityMarker = 4 -15
+         - filter: RejectList
+           where:
+           - variable: ObsType/airTemperature
+             is_in: 187
+           - variable: QualityMarker/airTemperature
+             is_in: 4-15
+
+         # Time window filter
+         - filter: Domain Check
+           where:
+           - variable:
+               name: MetaData/timeOffset
+             minvalue: -900
+             maxvalue:  900
+
+         # Temporal thinning filter
+         - filter: Temporal Thinning
+           min_spacing: PT30M
+           seed_time: *analysisDate
+           category_variable:
+             name: MetaData/stationIdentification
```

Comment:
@spanNOAA, are you finding that this is needed to match ob counts and increments? I found that for mesonet that this was actually detrimental for my results. Are you mainly using this to remove duplicate observations? If so, there could be other things we could try to handle that. I don't think the data is temporally thinned in GSI, so this "might" not be the best. Just something to consider.

Reply:
I found that removing duplicate METAR observations actually improved my results. In my case, eliminating them improved the analysis, as the duplicates may add additional value at the same location even if the information was outdated or from a significantly later time. 

I believe the detrimental results when applying temporal thinning to mesonet observations may be due to the smaller observation window used. For mesonet data, a ±6 minute window is applied, while for METAR data, a larger ±15 minute window is used. This is because some METAR stations have less frequent observations. As a result, applying a smaller window may cause some stations to be dropped, while others might have multiple observations within the ±15 minute window. 

Furthermore, GSI applies a form of "temporal thinning" when the l_closeobs option is enabled. This method retains only the observation closest to the analysis time and discards the others. Below is the relevant code:
```f90
if (l_closeobs) then
    if(abs(data(itime,k)-hr_offset)<abs(data(itime,l)-hr_offset)) then
        muse(l)=.false.
    else
        muse(k)=.false.
    endif
else
    tfact=min(one,abs(data(itime,k)-data(itime,l))/dfact1)
    dup(k)=dup(k)+one-tfact*tfact*(one-dfact)
    dup(l)=dup(l)+one-tfact*tfact*(one-dfact)
endif
```

Temporal thinning works well for temperature, humidity, and wind observations. However, I've had an issue with temporal thinning for surface pressure in my case. Specifically, surface observations in Canada were being dropped when temporal thinning was enabled. Theoretically, there should always be at least one observation at each station. I'm still investigating the cause of this issue.

30M spacing
![image](https://github.com/user-attachments/assets/f88b7723-1c99-4ff4-b416-0e0cbc706167)
0M spacing
![image](https://github.com/user-attachments/assets/7c8fdebc-8440-4580-bad5-61810ce2ae74)

Reply:
We should try to remove the duplicated observations from surface stations. @delippi Could you provide more details on how this filter detriment your MESONET results?

Reply:
Here are some runs I just made for mesonet using the temporal filter. The time window is +/-6 min. "No filter" gives better observation counts (counts are in the image; lower left corner), but increments look largely the same. The difference (between no filter and PT01M) is hard to spot but mainly just over west Texas. No temporal filter just seems better to me. PT30M definitely seems not good for this case. It could be argued that PT00M or PT01M would be fine.

**No Temporal filter**: (This gives different result from PT00M)
![fv3jedi_vs_gsi_increment_airTemperature_msonet_airTemperature_188_None](https://github.com/user-attachments/assets/3bf6fcdd-4b6e-48a9-9f41-22a57022b9da)
**PT01M**:
![fv3jedi_vs_gsi_increment_airTemperature_msonet_airTemperature_188_PT01M](https://github.com/user-attachments/assets/4efcf6f4-aa50-4281-8b37-b1193a156f41)
**PT30M**:
![fv3jedi_vs_gsi_increment_airTemperature_msonet_airTemperature_188_PT30M](https://github.com/user-attachments/assets/d71884ae-91c7-4dd5-b1f1-2028e425633e)